test/controllers/api/v2/hosts_controller_test.rb

```diff
@@ -1265,4 +1265,103 @@ def setup
+        "state" => "na",
+        "title" => "N/A"
+      }
+
+      get :power_status, params: { id: another_host.to_param }
+      assert_equal expected_response, JSON.parse(@response.body)
+    end
+
+    test 'show power status for a host' do
+      expected_resp = {
+        "id" => host.id,
+        "state" => "on",
+        "title" => "On"
+      }
+
+      Host.any_instance.stubs(:supports_power?).returns(true)
+      Host.any_instance.stubs(:supports_power_and_running?).returns(true)
+      get :power_status, params: { :id => host.id }, session: set_session_user, xhr: true
+      assert_response :success
+      response = JSON.parse @response.body
+      assert_equal(expected_resp.sort, response.sort)
+    end
+
+    test 'show power status for a powered off host' do
+      expected_resp = {
+        "id" => host.id,
+        "state" => "off",
+        "title" => "Off"
+      }
+
+      Host.any_instance.stubs(:supports_power?).returns(true)
+      Host.any_instance.stubs(:supports_power_and_running?).returns(false)
+      get :power_status, params: { :id => host.id }, session: set_session_user, xhr: true
+      assert_response :success
+      response = JSON.parse @response.body
+      assert_equal(expected_resp.sort, response.sort)
+    end
+
+    test 'show power status for a host that has no power' do
+      expected_resp = {
+        "id" => host.id,
+        "state" => "na",
+        "title" => 'N/A',
+        "statusText" => "Power operations are not enabled on this host."
+      }
+
+      Host.any_instance.stubs(:supports_power?).returns(false)
+      get :power_status, params: { :id => host.id }, session: set_session_user, xhr: true
+      assert_response :success
+      response = JSON.parse @response.body
+      assert_equal(expected_resp.sort, response.sort)
+    end
+
+    test 'show power status for a host that has an exception' do
+      expected_resp = {
+        "id" => host.id,
+        "state" => "na",
+        "title" => "N/A",
+        "statusText" => "Failed to fetch power status: ERF42-9958 [Foreman::Exception]: Unknown power management support - can't continue"
+      }
+
+      Host.any_instance.stubs(:supports_power?).returns(true)
+      Host.any_instance.stubs(:power).raises(::Foreman::Exception.new(N_("Unknown power management support - can't continue")))
+      get :power_status, params: { :id => host.id }, session: set_session_user, xhr: true
+      assert_response :success
+      response = JSON.parse @response.body
+      assert_equal(expected_resp.sort, response.sort)
+    end
+
+    test 'do not provide power state on an unknown host' do
+      get :power_status, params: { :id => 'no-such-host' }, session: set_session_user, xhr: true
+      assert_response :not_found
+    end
+
+    test 'do not provide power state for non ajax requests' do
+      get :power_status, params: { :id => host.id }, session: set_session_user
+      assert_response :method_not_allowed
```

Comment:
@ohad or @timogoebel is there any reason why method_not_allow is expected here ? I took this from the hosts_controller tests. I think 422 fits better here.

Reply:
I've been summoned, but I think you meant @ohadlevy ;).

Reply:
Ah whoops :D Sorry about that!

Reply:
the idea was that if its not AJAX it should not work, but in API context, I think you are right

Reply:
Ok cool then iI will leave it as `422` and adjust the test. Thanks for the info!